In [30]:
import pandas as pd
import numpy as np

AMP = '/kaggle/input/apartment-market-prices/Apartment_Market_Prices new.csv'
df = pd.read_csv(AMP)

list(df.columns)

['Year',
 'OBJECTID',
 'GEOID',
 'Tract Label',
 'Tract Name',
 'Community Reporting Area Name',
 'Community Reporting Area ID',
 'Tract Median Apartment Contract Rent per Square Foot',
 'Tract Median Apartment Contract Rent per Unit',
 'Year over Year Change in Rent per Square Foot',
 'Year over Year Change in Rent per Unit',
 'Cost Category',
 'Year over Year Change in Rent Category',
 'Mixed-Rate or Mixed-Income Apartments in Tract',
 'PROPERTIES',
 'Shape__Area',
 'Shape__Length']

In [21]:
df.shape

(4320, 17)

In [16]:
df.describe()

Year     OBJECTID         GEOID  Tract Label  \
count  4320.000000  4320.000000  4.320000e+03  4320.000000   
mean   2012.500000  2160.500000  5.303301e+10    66.461500   
std       6.922988  1247.220911  4.255099e+03    42.550986   
min    2001.000000     1.000000  5.303300e+10     1.010000   
25%    2006.750000  1080.750000  5.303300e+10    37.505000   
50%    2012.500000  2160.500000  5.303301e+10    67.015000   
75%    2018.250000  3240.250000  5.303301e+10    94.250000   
max    2024.000000  4320.000000  5.303303e+10   265.000000   

       Community Reporting Area ID  \
count                  4320.000000   
mean                      7.476111   
std                       3.471062   
min                       1.100000   
25%                       4.300000   
50%                       7.400000   
75%                      10.300000   
max                      13.300000   

       Tract Median Apartment Contract Rent per Square Foot  \
count                                        4320.000000      
mean                                            1.956067      
std                                             0.932163      
min                                             0.000000      
25%                                             1.510000      
50%                                             2.100000      
75%                                             2.600000      
max                                             4.220000      

       Tract Median Apartment Contract Rent per Unit  \
count                                    4320.000000   
mean                                     1289.829167   
std                                       624.631592   
min                                         0.000000   
25%                                      1050.000000   
50%                                      1352.000000   
75%                                      1653.000000   
max                                      3683.000000   

       Mixed-Rate or Mixed-Income Apartments in Tract   PROPERTIES  
count                                     4320.000000  4320.000000  
mean                                       547.108565    22.421296  
std                                        528.865123    21.058124  
min                                          0.000000     0.000000  
25%                                        122.000000     6.000000  
50%                                        396.000000    16.000000  
75%                                        852.000000    33.000000  
max                                       3708.000000   109.000000

In [19]:
df.head()

Year  OBJECTID        GEOID  Tract Label          Tract Name  \
0  2001         1  53033001201        12.01  Census Tract 12.01   
1  2001         2  53033000404         4.04   Census Tract 4.04   
2  2001         3  53033004901        49.01  Census Tract 49.01   
3  2001         4  53033003601        36.01  Census Tract 36.01   
4  2001         5  53033002800        28.00     Census Tract 28   

  Community Reporting Area Name  Community Reporting Area ID  \
0          Northgate/Maple Leaf                          8.1   
1         Broadview/Bitter Lake                          9.1   
2                       Fremont                         11.1   
3                    Green Lake                          9.4   
4       Greenwood/Phinney Ridge                          9.3   

   Tract Median Apartment Contract Rent per Square Foot  \
0                                               1.52      
1                                               1.29      
2                                               1.79      
3                                               2.10      
4                                               1.60      

   Tract Median Apartment Contract Rent per Unit  \
0                                           1085   
1                                           1042   
2                                           1042   
3                                           1197   
4                                            989   

  Year over Year Change in Rent per Square Foot  \
0                           482.758.620.689.656   
1                           403.225.806.451.613   
2                           529.411.764.705.883   
3                           552.763.819.095.478   
4                           526.315.789.473.685   

  Year over Year Change in Rent per Unit Cost Category  \
0                    513.565.891.472.868           Low   
1                                     42           Low   
2                    525.252.525.252.525        Medium   
3                    555.555.555.555.556        Medium   
4                    521.276.595.744.681           Low   

  Year over Year Change in Rent Category  \
0                                 Stable   
1                                 Stable   
2                                 Stable   
3                                 Stable   
4                                 Stable   

   Mixed-Rate or Mixed-Income Apartments in Tract  PROPERTIES  \
0                                            1209          21   
1                                             454           7   
2                                             977         106   
3                                             412          34   
4                                             290          31   

           Shape__Area        Shape__Length  
0   10.851.861.641.449  158.997.733.430.541  
1   67.916.527.237.854  129.625.584.131.713  
2  535.088.821.542.358  937.965.576.684.679  
3  725.587.906.799.316  118.942.888.948.854  
4  106.371.518.217.773   13.201.311.297.987

In [23]:
# Get the unique count for 'Community Reporting Area Name'
community_area_count = df['Community Reporting Area Name'].nunique()

# I'd also check the 'Tract Name' for a complete picture
tract_name_count = df['Tract Name'].nunique()

print(f"Total unique Community Reporting Areas: {community_area_count}")
print(f"Total unique Tract Names: {tract_name_count}")

Total unique Community Reporting Areas: 53
Total unique Tract Names: 180


# Types of Analysis
(based on who will value information)

1. Trend Analysis (Investors, Planners)
2. Geographic and Comparative Analysis (Investors, Renters)
3. Policy and Market Composition (City Planners)
4. Exploratory Data Analysis (EDA)

### General Insights
    1. What's the shape?
        - 4320 rows, and 17 columns
    2. What's the timeframe?
        - 2001 - 2024
    3. What's the data Quality?
        - Excellent, no nulls detected
    4. What's the Geographic Scope?
        - Total unique Community Reporting Areas: 53
        - Total unique Tract Names: 180

# Exploratory Data Analysis (EDA)
    1. Average Rent Price in Seattle
        - What's the "average" rent a person experiences?
        - Calculate the median for Tract Median Apartment Contract Rent per Unit and ...Rent per Square Foot across the whole dataset. (We use median, not mean, because rent prices are almost always skewed by a few super expensive properties)
        
    2 Market Moods / Trends
        - In any given year, what's the most likely Cost Category or Year over Year Change for a tract?
        - Get a value_counts() for the Cost Category and Year over Year Change in Rent Category columns. This will show us the distribution.
        
    3. Big Time Trend
        - How has rent in Seattle actually changed over the last 20+ years?
        - Group the data by Year and get the median Rent per Unit for each year. Then, plot this as a line chart. This is the single most important chart we can make.
        
    4. Most Least Expensive Areas
        - Where should I look if I'm rich? Where can I find a deal?
        - Filter the data for the most recent year (2024). Then, group by Community Reporting Area Name and find the median Rent per Unit. We can then sort this to find the Top 5 and Bottom 5.

In [12]:
# Avg Rent Price in Seattle
tmacr_psf = df['Tract Median Apartment Contract Rent per Square Foot'].median()
tmacr_pu = df['Tract Median Apartment Contract Rent per Unit'].median()

print(tmacr_psf)
print(tmacr_pu)

2.1
1352.0


In [39]:
# Market Mood / Trends
columns_to_fix = ['Cost Category','Year over Year Change in Rent Category']
df[columns_to_fix] = df[columns_to_fix].replace('0','Unknown')

print(df['Cost Category'].value_counts())
print(df['Year over Year Change in Rent Category'].value_counts())

Cost Category
Medium     1283
Low        1256
High       1256
Unknown     525
Name: count, dtype: int64
Year over Year Change in Rent Category
Stable              2607
Decline              720
Unknown              525
Significant Rise     352
Rise                 116
Name: count, dtype: int64


In [16]:
# Big Time Trend
grouped_by_year = df.groupby('Year')
print(grouped_by_year['Tract Median Apartment Contract Rent per Unit'].median().sort_values(ascending=False))

Year
2024    1729.0
2023    1719.0
2022    1715.0
2021    1655.5
2019    1592.5
2020    1570.5
2018    1550.0
2017    1530.0
2016    1485.0
2015    1440.5
2014    1373.5
2013    1333.5
2012    1284.5
2008    1277.0
2011    1254.0
2007    1251.0
2010    1235.5
2009    1193.5
2006    1170.0
2005    1095.0
2001    1084.5
2002    1056.0
2004    1056.0
2003    1039.5
Name: Tract Median Apartment Contract Rent per Unit, dtype: float64


In [33]:
# Most Least Expensive Areas
rent_cols = ['Tract Median Apartment Contract Rent per Unit', 
             'Tract Median Apartment Contract Rent per Square Foot']
df[rent_cols] = df[rent_cols].replace(0, np.nan)

grouped_by_area = df[df['Year'] == 2024].groupby('Community Reporting Area Name')
print("Top 5 Expensive Area: ")
print(grouped_by_area['Tract Median Apartment Contract Rent per Unit'].median().sort_values(ascending=False).head(5))
print("Top 5 Least Expensive Area: ")
print(grouped_by_area['Tract Median Apartment Contract Rent per Unit'].median().sort_values(ascending=True).head(5))


Top 5 Expensive Area: 
Community Reporting Area Name
Downtown Commercial Core      3099.5
Cascade/Eastlake              2688.0
Belltown                      2353.0
South Beacon Hill/NewHolly    2317.0
Duwamish/SODO                 2209.0
Name: Tract Median Apartment Contract Rent per Unit, dtype: float64
Top 5 Least Expensive Area: 
Community Reporting Area Name
South Park                 892.0
Laurelhurst/Sand Point    1237.0
Georgetown                1318.0
Wedgwood/View Ridge       1327.0
Riverview                 1333.0
Name: Tract Median Apartment Contract Rent per Unit, dtype: float64
